ethcore/src/engines/authority_round/mod.rs

```diff
@@ -1027,10 +1029,11 @@ impl Engine<EthereumMachine> for AuthorityRound {
 					result.map_err(|e| format!("{}", e))
 				};
 
-				c.reward(&benefactors, &mut call)?
+				let rewards = c.reward(&benefactors, &mut call)?;
+				rewards.into_iter().map(|(author, amount)| (author, RewardKind::External, amount)).collect()
```

Comment:
Unfortunately when using the block reward contract all rewards must be tagged as `External`, there's no way to appropriately map to benefactors list that was passed to the contract.